experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -439,7 +368,23 @@ export function makeOpGenerator(
 			sumWeights([passedOpWeights.abort, passedOpWeights.commit, passedOpWeights.start]),
 		],
 	];
-	return createWeightedAsyncGenerator(generatorWeights);
+
+	const generatorAssumingTreeIsSelected = createWeightedGenerator<Operation, EditState>(
+		generatorWeights,
+	);
+	return async (state) => {
+		// Even though not all ops require a client (e.g. synchronize), selecting one won't hurt.
+		// Centralizing this choice reduces boilerplate across nested generators and ensures all
+		// generators agree which client to perform the operation on.
+		// This selection doesn't include the last tree when there's mroe than one client, as having
+		// a passive client to compare against is convenient.
+		// TODO: If clients can join & leave mid-test, 'the last client' isn't stable with the natural
```

Comment:
not a problem now, but figured I'd call it out. Designating a client specifically in the state might be a more reliable mechanism.